tests/integration/test_create_token.py

```diff
@@ -0,0 +1,173 @@
+        "theme": "social",
+        "additional_payload": {
+            "flag_1": True,
+            "user_id": "abc",
+            "date": "2016-05-12",
+        },
+        "payload": {
+            "version": AuthPayloadVersion.V2.value,
+            "survey_metadata": {
+                "data": {
+                    "case_ref": "1000000000000001",
+                    "qid": str(uuid.uuid4()),
+                },
+                "receipting_keys": ["qid"],
+            },
+            "collection_exercise_sid": "789",
+            "response_id": "1234567890123456",
+            "language_code": "en",
+            "roles": [],
+            "account_service_url": ACCOUNT_SERVICE_URL,
+        },
+    },
+]
+
+
+class TestCreateToken(IntegrationTestCase, AppContextTestCase):
+    """
+    The purpose of this test class is to test the creation of a token (from create_token.py) to ensure
+    metadata in a decrypted token is nested/found in the correct level.
+    """
+
+    def test_payload_from_token(self):
+        for value in test_parameters:
+            with self.subTest():
+                additional_payload = value.get("additional_payload")
+                token = self.token_generator.create_token_v2(
+                    schema_name=value.get("schema"),
+                    theme=value.get("theme"),
+                    **additional_payload,
+                )
+                with self.test_app.app_context():
+                    decrypted_token = decrypt_token(token)
+                    self.assertEqual(
+                        decrypted_token, value.get("payload") | decrypted_token
+                    )
+                    assert (
+                        additional_payload.items()
+                        <= decrypted_token.get("survey_metadata").get("data").items()
+                    )
+
+    def test_uuid_consistent_after_decryption(self):
+        token = self.token_generator.create_token_v2(
+            "test_checkbox.json", theme="social", value="Dummy Text"
+        )
+        with self.test_app.app_context():
+            decrypted_token = decrypt_token(token)
+            assert decrypted_token.get("survey_metadata") == {
+                "data": {
+                    "case_ref": "1000000000000001",
+                    "qid": PAYLOAD_V2_SOCIAL.get("survey_metadata")
+                    .get("data")
+                    .get("qid"),
+                    "value": "Dummy Text",
+                },
+                "receipting_keys": ["qid"],
+            }
+
+    def test_sds_metadata_included_in_token(self):
+        token = self.token_generator.create_supplementary_data_token(
+            "test_checkbox.json"
+        )
+        with self.test_app.app_context():
+            decrypted_token = decrypt_token(token)
+            self.assertEqual(
+                decrypted_token, PAYLOAD_V2_SUPPLEMENTARY_DATA | decrypted_token
+            )
+
+    def test_structure_of_sds_metadata_in_token(self):
```

Comment:
I wouldn't say this is sds specific, just name the test after what you are exactly testing here, so something like `test_extra_payload(or metadata)_items_are(is)_added_to_token`. Similar to what you are doing below, where you call it: `test_metadata_is_removed_from_token`

Reply:
Previously, I had one test which was the combination of `test_structure_of_sds_metadata_in_token` and `test_sds_metadata_included_in_token`. I had a previous comment from Rhys where the name of the old test was confusing as it was testing two things so I split these tests up to make the purposes of the tests clear. I understand your point and I'm happy to rename but the intended purpose of this test was to ensure the structure of `survey_metadata` matched with the `survey_metadata` in `PAYLOAD_V2_SUPPLEMENTARY_DATA`.

Reply:
If I changed it, a more appropriate name could be `test_additional_payload_nested_in_token`?

Reply:
Yeah, this is a good and better name.

Reply:
Updated 👍